oak-run/src/main/java/org/apache/jackrabbit/oak/run/RevisionsCommand.java

```diff
@@ -74,15 +89,18 @@ public class RevisionsCommand implements Command {
 
     private static final Logger LOG = LoggerFactory.getLogger(RevisionsCommand.class);
 
+    private static final int REVISION_CAP = getInteger("oak.revision.cap", 250);
+
     private static final String USAGE = Joiner.on(System.lineSeparator()).join(
             "revisions {<jdbc-uri> | <mongodb-uri>} <sub-command> [options]",
             "where sub-command is one of",
             "  info     give information about the revisions state without performing",
             "           any modifications",
-            "  collect      perform garbage collection",
-            "  reset        clear all persisted metadata",
-            "  sweep        clean up uncommitted changes",
-            "  detailedGC   perform detailed garbage collection i.e. remove unmerged branch commits, old revisions, deleted properties etc"
+            "  collect        perform garbage collection",
+            "  reset          clear all persisted metadata",
+            "  sweep          clean up uncommitted changes",
+            "  detailedGC     perform detailed garbage collection i.e. remove unmerged branch commits, old revisions, deleted properties etc",
+            "  pathCleanup    clean up old/unused revisions and unmerged branch commits on a specific path"
```

Comment:
That's possible, yes, and would be cleaner IMO. Since the `--path` is already a separated argument we can simply apply it to this command.
My only concern with this approach is, if the `path` argument is missing, it could end up running the DetailedGC in all the repository.

Reply:
In theory if someone runs the command they should be aware of what it does - it's an admin command after all.

I guess we're doing it this careful because the new code isn't tested well enough yet.

Could we then - during this testing phase - demand that you either specify `--path` or `--entirerepo` (or something like this)? Then eventually make `--entirerepo` the default and not demand it anymore?

Reply:
That makes me a little less nervous. I have added that option for now.